packages/fast-usdc-contract/src/exos/advancer.ts

```diff
@@ -201,8 +207,22 @@ export const prepareAdvancerKit = (
             const { EUD } = decoded.query;
             log(`decoded EUD: ${EUD}`);
             assert.typeof(EUD, 'string');
-            // throws if the bech32 prefix is not found
+
+            // throws if it's neither CAIP-10 nor bare bech32.
             const destination = chainHub.resolveAccountId(EUD);
+            const destChain = chainOfAccount(destination);
+            const info = await when(chainHub.getChainInfoByChainId(destChain));
```

Comment:
isn't this synchronous?

```suggestion
            const info = chainHub.getChainInfoByChainId(destChain);
```

Reply:
fixed